src/MongoDB.Driver.Core/Core/Operations/FindOneAndUpdateOperation.cs

```diff
@@ -56,15 +53,12 @@ public class FindOneAndUpdateOperation<TResult> : FindAndModifyOperationBase<TRe
         /// <param name="update">The update.</param>
         /// <param name="resultSerializer">The result serializer.</param>
         /// <param name="messageEncoderSettings">The message encoder settings.</param>
-        public FindOneAndUpdateOperation(CollectionNamespace collectionNamespace, BsonDocument filter, BsonDocument update, IBsonSerializer<TResult> resultSerializer, MessageEncoderSettings messageEncoderSettings)
+        public FindOneAndUpdateOperation(CollectionNamespace collectionNamespace, BsonDocument filter, BsonValue update, IBsonSerializer<TResult> resultSerializer, MessageEncoderSettings messageEncoderSettings)
             : base(collectionNamespace, resultSerializer, messageEncoderSettings)
         {
             _filter = Ensure.IsNotNull(filter, nameof(filter));
             _update = Ensure.IsNotNull(update, nameof(update));
-            if (_update.ElementCount == 0)
-            {
-                throw new ArgumentException("Updates must have at least 1 update operator.", nameof(update));
-            }
+            ValidateUpdateValue(update);
```

Comment:
If you rename ValidateUpdateValue and refactor it appropriately you can replace the previous two lines with:

    _update = EnsureUpdateIsValid(update);